pkg/provider/cloud/vsphere/provider.go

```diff
@@ -119,8 +119,14 @@ func (v *VSphere) DefaultCloudSpec(ctx context.Context, spec *kubermaticv1.Clust
 		if v.dc.DefaultTagCategoryID != "" {
 			spec.Cloud.VSphere.Tags = &kubermaticv1.VSphereTag{
 				CategoryID: v.dc.DefaultTagCategoryID,
+				Tags:       []string{},
 			}
 		}
+	} else {
+		// If tags were specified without a CategoryID then use the default CategoryID, if configured at seed level.
```

Comment:
What if tags were specified without a CategoryID and no defaultCategoryID is specified/set on seed level?

Reply:
The validating webhook will reject such a cluster.

https://github.com/kubermatic/kubermatic/pull/11928/files#diff-4fa162a12c4d00d9041c47cdc45569b25718ef42bd5b748dc8b57838c34da641R151